### ISSUE TITLE
Doc for prefer-json-parse-buffer has wrong Pass suggestions (Typescript complains) 

### ISSUE THREAD

**Issue by aligatorr89** [sindresorhus/eslint-plugin-unicorn#1921]
Document https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v40.1.0/docs/rules/prefer-json-parse-buffer.md has wrong Pass suggestions. Typescript complains and solution is pretty simple. Use `toJSON()` method on returned buffer instead of trying with JSON.parse

**Comment by kachkaev** [comment]
Not sure about the proposed solution, but I have the same issue with Unicorn not playing nicely with TS:

```ts
const myJson = JSON.parse(fs.readFileSync(myFileName, "utf8"));
//                                                     ^^^^^
// 🚨 Prefer reading the JSON file as a buffer. eslint(unicorn/prefer-json-parse-buffer)
```

↑↓

```ts
const myJson = JSON.parse(fs.readFileSync(myFileName));
//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
// 🚨 Argument of type 'Buffer' is not assignable to parameter of type 'string'. ts(2345)
```

I’ve disabled `unicorn/prefer-json-parse-buffer` for now.

**Comment by aligatorr89** [comment]
Hey, sorry for late response @kachkaev ,. I had this problem, but never had time to create pull request. Anyway, seems like it's solved in v44.
